cEP-0027.md

```diff
@@ -0,0 +1,149 @@
+# Improve Lint Bear Quality
+
+| Metadata |                                           |
+| -------- | ----------------------------------------- |
+| cEP      | 27                                        |
+| Version  | 1.0                                       |
+| Title    | Improve Lint Bear Quality                 |
+| Authors  | Sangam Kumar <mailto:ksangam68@gmail.com> |
+| Status   | Proposed                                  |
+| Type     | Feature                                   |
+
+## Abstract
+
+This cEP describes the process of adding `BaseTestHelper` class and
+`GlobalBearTestHelper` class to improve the testing API of coala bears, as a
+part of the [GSoC 2018
+project](https://summerofcode.withgoogle.com/projects/#6625036551585792).
+
+## Introduction
+
+[coala](https://coala.io) has its own testing API which includes
+`LocalBearTestHelper`, a helper class for testing of local bears. It does not
+provide support for `GlobalBear`. Writing tests for a `GlobalBear` is a
+bit lengthy task for a developer. Example:
+[GitCommitBearTest](https://github.com/coala/coala-bears/blob/master/tests/vcs/git/GitCommitBearTest.py),
+[VultureBearTest](https://github.com/coala/coala-bears/blob/master/tests/python/VultureBearTest.py), etc.
+`GlobalBearTestHelper` is going to be an interface for the testing of
+`GlobalBear`.
+
+Also, the testing API should have a base test helper class which should be
+inherited in all tests' helper (LocalBearTestHelper, GlobalBearTestHelper).
+This project introduces a `BaseTestHelper` to unify unittest functionality.
+
+## Proposed Approach
+
+### 1. Implementation of `BaseTestHelper` class
+
+Here is the prototype implementation of `BaseTestHelper` class:
+
+- 1. `BaseTestHelper` to unify unittest:
+     ```
+     class BaseTestHelper(unittest.TestCase):
+         pass
+     ```
+     Inherit it in test helper API
+     ```
+     class LocalBearTestHelper(BaseTestHelper):
+     ```
+     And
+     ```
+     class GlobalBearTestHelper(BaseTestHelper):
+     ```
+
+- 2. Add `TestTimer` class inheriting `BaseTestHelper` class:
+
+     ```
+     class TestTimer(BaseTestHelper):
+         warning_time = 10
+
+         def setUp(self):
+             self.startedtime = time()
+
+         def tearDown(self):
+             self.completedtime = time()
+             duration = self.completedtime - self.startedtime
+             If duration > warning_time:
+                 print(duration)
+                 sys.stdout.flush()
+     ```
+
+### 2. Implementation of `GlobalBearTestHelper` class
+
+Currently, many methods are redundant in the Global Bear tests. For example:
```

Comment:
@li-boxuan, I don't think it needs a comparison. It has same difference as LocalBear and GlobalBear have. In GlobalBearTestHelper, it will help in testing a list of files.